app/controllers/games_controller.rb

```diff
@@ -37,6 +37,36 @@ def update
     @pieces = @game.pieces
   end
 
+  def castle_kingside
+    @game = Game.find(params[:game_id])
+    y = @game.white_player_id == current_user.id ? 0 : 7
+    @king = @game.pieces.find_by(:x_coord => 4, :y_coord => y) 
+    if @king && @king.can_castle?(7,y)
+      @rook = @game.pieces.find_by(:x_coord => 7, :y_coord => y)
```

Comment:
@bhk1019 what if we move the most logic inside the game model? the controller then would be only responsible for passing through the data and delegating everything to the model.. sth like:

```
def castle_kingside
    @game = Game.find(params[:game_id])
    if @game.can_castle?(current_user.id)
      @game.castle!(current_user.id)     
    else
      flash.now[:alert] = "You cannot castle at this time."
    end
    render :show
  end
```

The `can_castle?(user_id)` method on the game could be responsible for getting the right king and rook and doing all the needed checks..

You could even take it one step further:

```
def castle_kingside
    @game = Game.find(params[:game_id])
    if @game.castle!(current_user.id)  
       flash.now[:success] = "Castling successful!"
    else
      flash.now[:alert] = "You cannot castle at this time."
    end
    render :show
  end
```
In this option, `game.castle!(current_user.id) ` would call the `can_castle?` internally and return false if it can't and true after it does castle.. This way the controller stays really simple and does not involve any business logic